lib/spack/spack/filesystem_view.py

```diff
@@ -588,6 +588,17 @@ def purge_empty_directories(self):
                 except OSError:
                     pass
 
+    def purge_broken_links(self):
```

Comment:
I think there is some logic like this in `stage.py` -- could that be consolidated?

Reply:
The stage logic is different: it contains a function called `purge`, but that actually removes files that links point to in addition to removing the links themselves (via `remove_linked_tree`). This function is intended to only remove symlinks that are broken.

Reply:
See `remove_dead_links` and `remove_if_dead_link` in `llnl.util.filesystem`.  Stage uses at least `remove_if_dead_link`.

Reply:
Apparently nothing used `remove_dead_links`, so I moved all the generic filesystem logic to `llnl.util.filesystem` and consolidated it there to address this.